### PR TITLE
REST Repositories for Models

### PR DESCRIPTION
**Pull request by RyanWork**
I'm not sure if we should be committing the target folder, but I noticed Kshamina used a template and I just kind of assumed it was right??? Not sure on this one, no clear cut answer from what I could find on Google.

Also, I put the CustomizedRestMvcConfiguration.java file in the root package. I wasn't really sure where to put it. Any suggestions? Or is leaving it at root okay?

### CODE REVIEW

File: src/main/java/SYSC4806/CustomizedRestMvcConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    /**
+     * Override method to change the base path for REST calls
+     * @param config
+     */
+    @Override
+    public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
+        config.setBasePath("/api");
+    }
+}
+
```
Comment (SupriyaGadigone) [inline src/main/java/SYSC4806/CustomizedRestMvcConfiguration.java]: I think for now the location of this is okay here. I checked out some other repos and forums and people have been putting them either in the root or just the src package. So this should be fine.

**Review by SupriyaGadigone — APPROVED** [review]

**Review by adamstaples — APPROVED** [review]
What a legend, looks great!

**Review by andrewdlqnguyen — APPROVED** [review]
I cant seem to find anything online, besides some maven related implementations but from the look of the solutions, their suggestions are to put in src package. 
src/resource?  Think what you did is fine!

For the dependencies, could we put <version>RELEASE</version> under those that has without version? Just to keep the dependencies updated?

**Comment by RyanWork** [comment]
@SupriyaGadigone I just removed the original rerouting to the API and added a shorter thing to application.properties instead. :^) Probably can just accept it, but Kshamina told me to review EVVVVVVVVVVVERYTHING

**Review by SupriyaGadigone — APPROVED** [review]

**Pull request merged**